python/dgl/sampling/neighbor.py

```diff
@@ -129,18 +128,24 @@ def sample_neighbors(g, nodes, fanout, edge_dir='in', prob=None, replace=False,
         else:
             nodes_all_types.append(nd.array([], ctx=nd.cpu()))
 
-    if not isinstance(fanout, dict):
-        fanout_array = [int(fanout)] * len(g.etypes)
+    if isinstance(fanout, nd.NDArray):
```

Comment:
why is `fanout` `nd.NDArray`?

Reply:
`_CAPI_DGLSampleNeighbors()` takes an `nd.NDArray` as the argument for fanout. Creating it each step is expensive relative to sampling, so I added the option in `MultiLayerNeighborSampler()` to construct it once, and keep passing it in here each step.